src/depackers/unxz.c

```diff
@@ -91,9 +91,9 @@ static int decrunch_xz(HIO_HANDLE *in, FILE *out, long inlen)
 		goto err;
 	*/
 
-	fwrite(buf.out, 1, buf.out_pos, out);
+	*out = buf.out;
+	*outlen = buf.out_pos;
```

Comment:
You should probably uncomment the above code to shrink the buffer first (I had a modified version of it that I didn't force push in time before it got merged):

```C
if ((tmp = (uint8 *) realloc(buf.out, buf.out_pos)) != NULL)
  buf.out = tmp;
```